src/mscorlib/src/Internal/Runtime/Augments/EnvironmentAugments.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Internal.Runtime.Augments
+{
+    /// <summary>For internal use only.  Exposes runtime functionality to the Environments implementation in corefx.</summary>
+    public static class EnvironmentAugments
+    {
+        public static int CurrentManagedThreadId => Environment.CurrentManagedThreadId;
+        public static void Exit(int exitCode) => Environment.Exit(exitCode);
+        public static int ExitCode { get { return Environment.ExitCode; } set { Environment.ExitCode = value; } }
```

Comment:
I'll add a test for it in corefx.  If it doesn't work I'll open an issue on that.  I don't think we'd change this augments implementation based on that, but rather would fix in the runtime whatever we'd need to get it working.  There's already a corefx test verifying that this value roundtrips, but that doesn't mean it actually ends up propagating the value out of the process.